returnn/tf/util/basic.py

```diff
@@ -59,6 +59,21 @@ def have_min_tf_version(version):
   return tf_version >= version
 
 
+def get_checkpoint_filepattern(filepath):
+  """
+  Removes optional .index or .meta extension
+
+  :param str filepath:
+  :return: CheckpointLoader compatible filepattern
+  :rtype: str
+  """
+  if filepath.endswith(".meta"):
+    return filepath[:-len(".meta")]
+  elif filepath.endswith(".index"):
+    return filepath[:-len(".index")]
+  return filepath
```

Comment:
As this is also used in engine base logic, it needs to be in the base utils, not TF specific utils.

See also function `get_model_filename_postfix`, which is related. Maybe put it below.